### DISCUSSION TITLE
How to pass variable arguments in gauge

### DISCUSSION THREAD

**Discussion by AlexPavl** [getgauge/gauge#2016] [General]
Hello there, i have a problem, don't know how to pass several, non constant, arguments in @step. I used before robot framework and there is a dictionary is acceptable option, but can't find anything suitable with gauge

**Comment by sriv** [comment]
If I understand correctly, you seek a way to pass arguments to the step implementation method? `@step` is an annotation, which can take arguments, but it is specific to gauge, e.x. the text it matches, tags etc.

To pass arguments to the implementation method, you can add those to the method signature and these will be substituted by the arguments in the step text.

Ex: in the sample, [there is a step](https://github.com/getgauge/template-java/blob/master/specs/example.spec#L10) `Vowels in English language are "aeiou".` Here, the value "aeiou" is passed as a parameter to [the implementation](https://github.com/getgauge/template-java/blob/master/src/test/java/StepImplementation.java#L13).

While there is no option to pass a dictionary, there are two things you can do:

- pass a table. See [this step in the sample](https://github.com/getgauge/template-java/blob/master/specs/example.spec#L25) for an example
- alternatively, you can roll out your mechanism by serializing the dictionary into a json (or any other format) and pass it to the implementation.

Hope this helps.

**Reply by AlexPavl** [reply]
Thank you, i thought about using tables for this purpose, but idea of using json i like more.
